sci-visualization/xd3d/xd3d-8.3.1-r1.ebuild

```diff
@@ -1,56 +1,54 @@
-# Copyright 1999-2012 Gentoo Foundation
+# Copyright 1999-2017 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id$
 
-EAPI=4
+EAPI=6
 
-inherit eutils fortran-2 multilib toolchain-funcs
+inherit fortran-2 toolchain-funcs
 
 DESCRIPTION="Scientific visualization tool"
 HOMEPAGE="http://www.cmap.polytechnique.fr/~jouve/xd3d/"
 SRC_URI="mirror://gentoo/${P}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="amd64 ppc ppc64 x86 ~amd64-linux ~x86-linux"
-IUSE="examples"
+KEYWORDS="~amd64 ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="doc examples"
 
 RDEPEND="
 	x11-libs/libXpm"
 DEPEND="${RDEPEND}
 	app-shells/tcsh"
 
+PATCHES=(
+	"${FILESDIR}"/${P}-gentoo.diff
+	"${FILESDIR}"/${P}-parallel.patch
+	"${FILESDIR}"/${P}-rotated.patch
+	"${FILESDIR}"/${P}-cflags.patch
+)
+
 src_prepare() {
-	epatch \
-		"${FILESDIR}"/${P}-gentoo.diff \
-		"${FILESDIR}"/${P}-parallel.patch \
-		"${FILESDIR}"/${P}-rotated.patch \
-		"${FILESDIR}"/${P}-cflags.patch
+	default
 	sed \
 		-e 's:"zutil.h":<zlib.h>:g' \
 		-i src/qlib/timestuff.c || die
-}
-
-src_configure() {
-	tc-export CC
 	sed \
-		-e "s:##D##:${ED}:" \
+		-e "s:##D##:${ED%/}:" \
 		-e "s:##lib##:$(get_libdir):" \
 		-i RULES.gentoo \
 		|| die "failed to set up RULES.gentoo"
+}
+
+src_configure() {
+	tc-export CC
 	./configure -arch=gentoo || die "configure failed."
 }
 
 src_install() {
 	dodir /usr/bin
-	emake install
-
-	dodoc BUGS CHANGELOG FAQ FORMATS README
-	insinto /usr/share/doc/${PF}
-	doins Manuals/*
+	default
 
-	if use examples; then
-		insinto /usr/share/doc/${PF}/
-		doins -r Examples
-	fi
+	#	dodoc BUGS CHANGELOG FAQ FORMATS README
+	use doc && dodoc -r Manuals
+	use examples && dodoc -r Examples
```

Comment:
again, consider `docompress -x` (I would also install into lowercase **examples** to be more in line with the rest of Gentoo)